File: manipulacion-de-texto/transformar_analizar_strings.py
texto: str = "  Aprendiendo Python es Divertido  "


# TODO: Elimina espacios con strip()
espaciosEliminados: str = texto.strip(" ")

# TODO: Convierte a minúsculas y mayúsculas
aMinusculas: str = espaciosEliminados.lower()
aMayusculas: str = aMinusculas.upper()

# TODO: Reemplaza "Python" por "Programación" con replace()
reemplazar: str = espaciosEliminados.replace("Python", "Programación")
print(reemplazar)

# TODO: Verifica si contiene "divertido" usando 'in' (case insensitive)
verificar: bool = "divertido" in reemplazar.lower()
print(f"Existe??? {verificar}")

# TODO: Cuenta vocales usando comprensión de listas
invertida: str = espaciosEliminados[:: -1]

# TODO: Invierte el string usando slicing [: :-1]
extraerConSplit: list = espaciosEliminados.split()
print(extraerConSplit)
# TODO:  Extrae palabras usando split() y slicing


extraerConSlicing: str = espaciosEliminados[0:11]
print(extraerConSlicing)
vocales: list = ["a", "e", "i", "o", "u"]
contarVocales = [letra for letra in espaciosEliminados.lower()
                 if letra in vocales]
print(contarVocales)
print(f"Cantidad de vocales: {len(contarVocales)}")

File: manipulacion-de-texto/transformar_analizar_strings.ts

const texto: string = "  Aprendiendo JavaScript es Divertido  ";

// TODO: Elimina espacios con trim()
const espaciosEliminados: string = texto.trim();
console.log(espaciosEliminados);
// TODO: Convierte a minúsculas y mayúsculas
const aMayusculas: string = espaciosEliminados.toLocaleUpperCase();
const aMinusculas: string = espaciosEliminados.toLocaleLowerCase();
console.log(aMayusculas);
// TODO:  Reemplaza "JavaScript" por "Programación" con replace()
const reemplazar: string = espaciosEliminados.replace("Javascript", "Programación");
// TODO: Verifica si incluye "divertido" (case insensitive)
const verifica: boolean = reemplazar.includes("divertido");

// TODO: Cuenta vocales usando match() con regex
const contarVocales = reemplazar.match(/[aeiou]/gi);
console.log(contarVocales ? contarVocales.length : 0);
// TODO: Invierte el string usando split(), reverse() y join()
const cortarTexto: string[] = reemplazar.split(" ");
const invertirConReverse: string[] = cortarTexto.reverse();
const unir: string = invertirConReverse.join(' ');
console.log(unir);
// TODO: Extrae palabras usando slice() o substring()
const extraer = unir.slice(0,9);
console.log(extraer)
const extraerconSub = unir.substring(0,9);
console.log(extraerconSub)